src/main/scala/Dispatcher.scala

```diff
@@ -0,0 +1,41 @@
+import scala.io.Source
+
+import telegram.API._
+import tinkoff.API._
+
+/**
+  * Main polling loop.
+  */
+object Dispatcher extends App {
+
+  def getFormattedRates(rates: List[Rate]): String = {
+    val textRatesList = rates.map(x => "%3s: %7.3f  %7.3f".format(x.fromCurrency.name, x.buy, x.sell.get))
+    "<pre>" +
+      "       Buy      Sell\n" +
+      "      ------   ------\n" +
+      textRatesList.mkString("\n") + "</pre>"
+  }
+
+  setToken(Source.fromFile("TelegramBotToken").getLines.mkString)
+
+  while (true) {
+    val updatesList = getUpdates
+
+    for (update <- updatesList; if update.message.isDefined; message = update.message.get) {
+        if (message.entities.isDefined)
+          for (entity <- message.entities.get if entity.`type` == "bot_command") {
+            // (jusual): message.text is defined because text contains name of bot_command
+            val command = message.text.get.substring(entity.offset, entity.offset + entity.length)
```

Comment:
You cannot apply map(substring) to String. Anyway, we have message entities where we know exact positions and lengths of every command.